src/main/scala/com/github/vitalsoftware/scalaredox/client/RedoxTokenManager.scala

```diff
@@ -0,0 +1,102 @@
+package com.github.vitalsoftware.scalaredox.client
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.model.Uri
+import akka.http.scaladsl.model.Uri.Path./
+import akka.stream.Materializer
+import org.joda.time.DateTime
+import play.api.libs.json.Json
+
+import scala.collection.concurrent.TrieMap
+import scala.concurrent.duration._
+import scala.concurrent.ExecutionContext.Implicits._
+import scala.concurrent.Future
+
+/**
+ * Redox uses a perishable accessToken with an expiry to authenticate requests. The accessToken is obtained by
+ * exchanging a apiKey and an apiSecret. On the first exchange we receive:
+ *  1. accessToken
+ *  2. expiry time
+ *  3. refreshToken
+ *
+ *  We use the RefreshToken obtained from the first request to obtain new accessTokens as they expire.
+ *
+ *  The RedoxTokenManager schedules the refresh and storage of accessTokens and keeps them valid,
+ *  so a client don't have to authenticate each time it makes a request.
+ *
+ *  @param client com.github.vitalsoftware.scalaredox.client.HttpClient instance
+ *  @param baseRestUri the base url for Redox REST endpoint.
+ */
+class RedoxTokenManager(
+  client: HttpClient,
+  baseRestUri: Uri
+)(implicit
+  actorSystem: ActorSystem,
+  materializer: Materializer
+) extends RedoxClientComponents(client, baseRestUri, identity) {
+
+  private val RefreshBuffer = 1.minute
+
+  /**
+   * Maps an apiKey to fetched AuthInfo.
+   */
+  private val TokenStore: TrieMap[String, AuthInfo] = TrieMap.empty
+
+  /**
+   * Returns an access token for given credentials. If the access token doesn't exist,
+   * authorize with Redox to obtain a new access token.
+   */
+  def getAccessToken(apiKey: String, apiSecret: String): Future[AuthInfo] = {
+    TokenStore.get(apiKey).map(Future.successful(_))
+      .getOrElse {
+        val authInfo = authorize(apiKey, apiSecret)
+        authInfo.foreach(setAuthAndScheduleRefresh(apiKey, _))
+        authInfo
+      }
+  }
+
+  /** Authorize to Redox, returning a Future containing the access and refresh tokens. */
+  protected def authorize(apiKey: String, apiSecret: String): Future[AuthInfo] = {
```

Comment:
This seems to do authentication, but the method name is authorize (authorization and authentication are different)